armi/materials/mgO.py

```diff
@@ -22,12 +22,28 @@
 class MgO(Material):
     r"""MagnesiumOxide"""
     name = "MgO"
+    propertyValidTemperature = {
+        "density": ((273, 1273), "K"),
+        "linear expansion percent": ((273.15, 1273.15), "K"),
+    }
 
     def setDefaultMassFracs(self):
         r"""mass fractions"""
         self.setMassFrac("MG", 0.603035897)
         self.setMassFrac("O16", 0.396964103)
 
+    def density(self, Tk=None, Tc=None):
+        """same reference as linear expansion. Table II.
+        Reference density is from Wolfram Alpha At STP (273 K)"""
+        Tk = getTk(Tc, Tk)
+        rho0 = 3.58
+        self.checkPropertyTempRange("density", Tk)
+        dLL = self.linearExpansionPercent(Tk=Tk)
+
+        dRho = (1 - (1 + dLL) ** 3) / (1 + dLL) ** 3
```

Comment:
I understand this is not the fault of this PR, but I'm noticing here that we have a number of density methods which return the expanded density in 3D. Here, for SiC, and I think other materials as well.

This is directly in contradiction with the base method docstring, which says the `density()` method returns the density as expanded in 2D: https://github.com/terrapower/armi/blob/main/armi/materials/material.py#L337

I think we probably need to escalate this issue to somewhere where we can track this for all of the material classes in the framework.

Reply:
This appears to have been resolved in #868 .